### PR TITLE
docs: add contributing guide for docs

### PR DESCRIPTION
**Pull request by charislam**
## What kind of change does this PR introduce?

**Docs**

Adds a contributing/style guide for docs to make the contributing process easier, to enforce consistent style and formatting guidelines, and to make our docs clear and readable.

### CODE REVIEW

File: apps/docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+Here are some general guidelines on writing docs for Supabase.
+
+## General principles
+
+Docs should be helpful, quick to read, and easy to understand. We have an audience of global readers who speak different native languages.
+
+To make docs as clear as possible:
+
+- Write for the user. Think about what task they want to complete by reading your doc. Tell them what, and only what, they need to know.
+- Write like you talk. Use words and sentences that sound natural when speaking. Cut out unnecessary words. Read your writing out loud to help you choose the clearest and simplest phrases.
```
Comment (kiwicopple) [inline apps/docs/CONTRIBUTING.md:16]:
```suggestion
- Write like you talk. Use words and sentences that sound natural when speaking. Cut unnecessary words. Read your writing out loud to help you choose the clearest and simplest phrases.
```

File: apps/docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+
+Use admonitions sparingly. Don't stack them on top of each other.
+
+Choose the appropriate `type` for your admonition:
+
+- **Danger** to warn the user about any missteps that could cause data loss or data leaks
+- **Deprecation** to notify the user about features that are (or will soon be) deprecated
+- **Caution** to warn about anything that could cause a bug or serious user inconvenience
+- **Tip** to point out helpful but optional actions
+- **Note** for anything else
```
Comment (kiwicopple) [inline apps/docs/CONTRIBUTING.md:166]:
```suggestion
- `danger` to warn the user about any missteps that could cause data loss or data leaks
- `deprecation` to notify the user about features that are (or will soon be) deprecated
- `caution` to warn about anything that could cause a bug or serious user inconvenience
- `tip` to point out helpful but optional actions
- `note` for anything else
```

File: apps/docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+
+Don't use blockquotes.
+
+### Code blocks
+
+Keep code lines short to avoid scrolling. For example, you can split long shell commands with `\`.
+
+- **JavaScript/TypeScript**
+
+  The `supabase` repo uses Prettier, which also formats JS/TS in code blocks. Your PR is blocked from merging if the Prettier check fails. Ensure that your code blocks are formatted by running `npm run format`, or by setting up auto-formatting in your IDE.
```
Comment (kiwicopple) [inline apps/docs/CONTRIBUTING.md:186]:
```suggestion
The `supabase` repo uses Prettier, which also formats JS/TS in code blocks. Your PR is blocked from merging if the Prettier check fails. Ensure that your code blocks are formatted by running `npm run format`, or by setting up auto-formatting in your IDE.
```

File: apps/docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+
+- **SQL**
+
+  Prefer lowercase for SQL. For example, `select * from table` rather than `SELECT * FROM table`.
```
Comment (kiwicopple) [inline apps/docs/CONTRIBUTING.md:190]:
```suggestion
Prefer lowercase for SQL. For example, `select * from table` rather than `SELECT * FROM table`.
```

**Review by kiwicopple — APPROVED** [review]

File: apps/docs/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+  - Labels of UI elements are bolded. For example, `Click **Confirm**.`
+
+### Word list
+
+- `Backend` isn't hyphenated (not `back-end`).
+- `Frontend` isn't hyphenated (not `front-end`).
+- `Login` is a noun. `Log in` is a verb.
+- `Postgres` is capitalized, except in code, and used instead of `PostgreSQL`.
+- `Setup` is a noun. `Set up` is a verb.
+- `Supabase` is always (not `supabase`), except in code.
```
Comment (charislam) [inline apps/docs/CONTRIBUTING.md:322]:
```suggestion
- `Supabase` is capitalized (not `supabase`), except in code.
```

File: apps/docs/CONTRIBUTING.md (hunk 3 above)
Comment (charislam) [reply apps/docs/CONTRIBUTING.md:186]: intentional to indent as part of list item

File: apps/docs/CONTRIBUTING.md (hunk 4 above)
Comment (charislam) [reply apps/docs/CONTRIBUTING.md:190]: intentional to indent as part of list item

**Pull request merged**